phpunit/functional/DBmysqlIteratorTest.php

```diff
@@ -160,18 +140,13 @@ function () {
      */
     public function testNoTableWithoutWhereBis()
     {
-        $this->exception(
-            function () {
-                $it = $this->it->execute(['FROM' => []]);
-                $this->string('SELECT *', $it->getSql(), 'No table');
-            }
-        )->isInstanceOf(\LogicException::class)
-         ->hasMessage('Missing table name.');
+        $this->expectExceptionObject(new \LogicException('Missing table name.'));
+        $this->it->execute(['FROM' => []]);
     }
 
-
-    public function testDebug()
+    /*public function testDebug()
     {
+        //cause issues on all other tests.. cannot unset or change a constant, I do not know how to proceed.
```

Comment:
If someone has an idea

Reply:
Not important to test `GLPI_SQL_DEBUG` at all. It only exists in 10.0 for BC but never read from AFAIK. We use the new Debug Bar/Debug Profile feature.

Reply:
That indeed does not seem important, I just don't like removing tests cases because of framework change - the idea is to keep it if possible :)

Reply:
Actually, I was wrong, `GLPI_SQL_DEBUG` still exists in `main`. All of the other SQL constants/globals were removed. If we keep the ability to log every SQL request to a file, it should be redone even in 10.0 to not use a constant at all. I don't think a breaking change for something so obscure and unimportant is a big deal.

Reply:
Did you tried `@runInSeparateProcess` ?

Reply:
adding that on method fail with `Serialization of 'Closure' is not allowed`

Reply:
You can do the test passing the `$debug = true` parameter to `$this->it->execute()`, but I guess it is not possible to do a test based on the constant value.

Reply:
OK for me like that, done in latest commit.

Reply:
You need to also add `@preserveGlobalState disabled` (don't ask me why, I just saw another test with the same tags).